Modules/timemodule.c

```diff
@@ -2390,3 +2398,44 @@ pysleep(PyTime_t timeout)
     return -1;
 #endif
 }
+
+
+#ifndef MS_WINDOWS
+// time.sleep(0) optimized implementation.
+// On error, raise an exception and return -1.
+// On success, return 0.
+static int
+pysleep_zero_posix(void)
+{
+    static struct timeval zero = {0, 0};
+
+    PyTime_t deadline, monotonic;
+    if (PyTime_Monotonic(&monotonic) < 0) {
+        return -1;
+    }
+    deadline = monotonic;
+    do {
+        int ret, err;
+        Py_BEGIN_ALLOW_THREADS
+        ret = select(0, (fd_set *)0, (fd_set *)0, (fd_set *)0, &zero);
```

Comment:
FTR: [`select(2)`](https://man7.org/linux/man-pages/man2/select.2.html) can mutate the timeout argument but if both fields are zero it immediately returns.